src/App.jsx

```diff
@@ -14,45 +17,85 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_FIELD_AZ = 'az';
+const SORT_FIELD_LENGTH = 'length';
+const SORT_FIELD_REVERSE = 'reverse';
+const SORT_FIELD_RESET = '';
+
+function getSortedGoods(goods, type) {
+  const sortedGoods = [...goods];
+
+  if (type[0] === SORT_FIELD_AZ) {
+    sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
+
+  if (type[0] === SORT_FIELD_LENGTH) {
+    sortedGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (type[1] === SORT_FIELD_REVERSE) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+export const App = () => {
+  const [sortType, setSortType] = useState(['', '']);
+  const sortedGoods = getSortedGoods(goodsFromServer, sortType);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={
+            cn('button is-info', { 'is-light': sortType[0] !== SORT_FIELD_AZ })
+          }
+          onClick={() => setSortType([SORT_FIELD_AZ, sortType[1]])}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn('button is-success',
+              { 'is-light': sortType[0] !== SORT_FIELD_LENGTH })
+          }
+          onClick={() => setSortType([SORT_FIELD_LENGTH, sortType[1]])}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={
+            cn('button is-warning',
+              { 'is-light': sortType[1] !== SORT_FIELD_REVERSE })
+          }
+          onClick={() => setSortType(
+            sortType[1] === SORT_FIELD_REVERSE
+              ? [sortType[0], SORT_FIELD_RESET]
+              : [sortType[0], SORT_FIELD_REVERSE],
+          )}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)